grpc/errors.go

```diff
@@ -96,7 +98,12 @@ func unwrapError(err error, md metadata.MD) error {
 					unwrappedErr,
 				)
 			}
-			outErr = (outErr.(*berrors.BoulderError)).WithSubErrors(suberrs)
+			var berr *berrors.BoulderError
+			if errors.As(outErr, &berr) {
+				outErr = berr.WithSubErrors(suberrs)
+			} else {
+				return fmt.Errorf("expected type of outErr to be %T got: %q", berr, outErr.Error())
```

Comment:
Note here you're logging "expected <type>, got <value>". The mismatch may be a bit confusing.

Reply:
Good call, I'll fix it.